peekaboo/server.py

```diff
@@ -274,6 +283,11 @@ def handle_requests(self, requests):
                 self.talk_back(json.dumps({'request': 'ping',
                                            'response': 'pong'}))
                 logger.debug('ping request received and answered')
+            elif reqtype == 'shutdown':
```

Comment:
@jack28: Here it occurs to me that as it stands every client could shut down the daemon. I expect there to be cases where we don't trust clients enough to be able to do that. IMO we wouldn't even want amavis to be able to do that. I can think of these solutions:
- don't add the function for now,
- have a separate command socket for the util,
- add authentication, maybe just a cookie or admin password from peekaboo.conf for now

What do you think?

BTW: I mainly added this to get an idea how to go about it. I'm not married to it.

Reply:
I favour some sort of token.

- kill embed mode
- json protocol to client
- move to nework socket (include samples in request)
- token to authorise (then new) advanced funtionality

Reply:
Agreed.